sdk/keyvault/azure-keyvault-keys/tests/_keys_test_case.py

```diff
@@ -0,0 +1,36 @@
+import os
+
+import pytest
+from devtools_testutils import AzureRecordedTestCase
+
+
+class KeysTestCase(AzureRecordedTestCase):
+    def _get_attestation_uri(self):
+        playback_uri = "https://fakeattestation.azurewebsites.net"
+        if self.is_live:
+            real_uri = os.environ.get("AZURE_KEYVAULT_ATTESTATION_URL")
+            real_uri = real_uri.rstrip('/')
+            if real_uri is None:
+                pytest.skip("No AZURE_KEYVAULT_ATTESTATION_URL environment variable")
+            return real_uri
+        return playback_uri
+
+    def create_crypto_client(self, key, **kwargs):
+        if kwargs.pop("is_async", False):
+            from azure.keyvault.keys.crypto.aio import CryptographyClient
+            credential = self.get_credential(CryptographyClient,is_async=True)
+        else:
+            from azure.keyvault.keys.crypto import CryptographyClient
+            credential = self.get_credential(CryptographyClient)
+
+        return self.create_client_from_credential(CryptographyClient, credential=credential, key=key, **kwargs)
+
+    def create_key_client(self, vault_uri, **kwargs):
```

Comment:
It looks like this method is only used in the single `test_parse_id.py` live test -- I wonder if we should restructure that test to use a client preparer instead and remove this method. It probably wouldn't hurt to test `test_parse_key_id_with_version` against all API versions; I wouldn't expect the behavior to be API version-dependent, but technically it depends on a response from the service so it's not unreasonable